From bc1ebdd7fd9fb219edc4fe9a92b2299fe0d3ae3b Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:27 -0400
Subject: [kernel] const: Add _BITUL() and _BITULL()

Message-id: <1406040016-3289-173-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86251
O-Subject: [RHEL7 PATCH v2 172/221] linux/const.h: Add _BITUL() and _BITULL()
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "H. Peter Anvin" <hpa@linux.intel.com>

Add macros for single bit definitions of a specific type.  These are
similar to the BIT() macro that already exists, but with a few
exceptions:

1. The namespace is such that they can be used in uapi definitions.
2. The type is set with the _AC() macro to allow it to be used in
   assembly.
3. The type is explicitly specified to be UL or ULL.

Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>
Link: http://lkml.kernel.org/n/tip-nbca8p7cg6jyjoit7klh3o91@git.kernel.org
(cherry picked from commit 2fc016c5bd8aad2e201cdf71b9fb4573f94775bd)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/uapi/linux/const.h b/include/uapi/linux/const.h
index c22c707..c872bfd 100644
--- a/include/uapi/linux/const.h
+++ b/include/uapi/linux/const.h
@@ -21,4 +21,7 @@
 #define _AT(T,X) ((T)(X))
 #endif
 
+#define _BITUL(x) (_AC(1,UL) << (x))
+#define _BITULL(x) (_AC(1,ULL) << (x))
+
 #endif /* !(_LINUX_CONST_H) */
-- 
1.7.1